internal/extproc/router/router.go

```diff
@@ -15,6 +15,9 @@ import (
 	"github.com/envoyproxy/ai-gateway/filterapi/x"
 )
 
+// ErrNoMatchingRule is the error when no matching rule is found.
+var ErrNoMatchingRule = errors.New("no matching rule found")
```

Comment:
I think this should be moved to the `filterapi/x` package (see my comment: https://github.com/envoyproxy/ai-gateway/pull/396#discussion_r1964826373).
Reason is that the concrete error is now part of the business logic of the chat completion processor and unless we move it there custom implementations of `Calculate` can't return it.

cc @mathetake

Reply:
https://github.com/envoyproxy/ai-gateway/pull/401